### ISSUE TITLE
Confusion regarding timing CORI and GRAV samples in GoPro MAX

### ISSUE THREAD

**Issue by DorSSS** [gopro/gpmf-parser#125]
Hey,
I'm having some difficulties with figuring out the timestamp of each CORI and GRAV sample in 360 regular and TimeLapse videos.
I have 2 cameras: one with the v1.5 firmware and one with the newer v2.0 firmware.
In the v1.5 firmware I had the following data:

- In timelapse with a frame every 0.5s - each sample had 1 value and the TSMP starting from 1 and incrementing by 1 with each sample
- in the regular video - each sample had 30 values besides the last and also the STMP(which seems more relevant for this) value starting from around ~150000 and increment by around 100000 each sample

From this, I understood the following logic - If in timelapse - the timestamp is (TSMP - 1) * 500ms, in a regular video the timestamp is STMP / 1000 + (value_in_sample_index  * 1 / 29.97 * 1000).
This logic works well for me when trying to rotate each image by the CORI value in order to get stabilized images.

In v2.0 the situation is a bit different from the timelapse mode and looks a bit more like the regular video.
Each sample has 1 value and besides the TSMP it now also has an STMP value starting from ~550000 and increment by 500000 each sample.
If applying the same logic from before it means that the first sample is actually of the second frame and that is everything is by and offset of 1 frame since TSMP=1(frame 0) has STMP of 500000.
This yields wrong results and the good results are generated if assigned the frame in offset 0 the sample with the STMP with the value of 550000 which is strange to me.

I try to read again the README and I saw GPMF Timing and Clocks sections that I should get the average sample rate and compare that against the video duration but here I got some strange results in some videos:

- The overall file duration_ms is the same as the first video track duration_ms
- The duration_ms of the second video track is sometimes less(usually by a frame in timelapse) then the first track duration_ms
- For example, I get 60 CORI samples, but the output of GoPro player MP4 has only 58 frames and if I do the stitching myself I get 59 frames which match with the second track duration_ms - obviously since its the shorter one and we need both of the to generate a full frame.

To summarize everything - I'm confused by the results in V2.0 and not sure what I'm missing or misunderstanding.
I would love to get some clarification about how this should be handled in both 360 timelapse videos and regular 360 videos.

**Comment by dnewman-gpsw** [comment]
CORI, IORI and GRAV, do not need to have their sample rate computed, and they are written differently to ensure there is one sample per frame encoded.   So the first CORI/GRAV sample applies to the first frame.
